packages/backend/src/accounting/psql/service.ts

```diff
@@ -194,15 +216,279 @@ export async function getSettlementBalance(
 
 export async function createTransfer(
   deps: ServiceDependencies,
-  {
+  args: TransferOptions
+): Promise<Transaction | TransferError> {
+  const sourceAccountRef = args.sourceAccount.id
+  const sourceAssetRef = args.sourceAccount.asset.id
+  const sourceAccountLedger = args.sourceAccount.asset.ledger
+
+  const destinationAccountRef = args.destinationAccount.id
+  const destinationAssetRef = args.destinationAccount.asset.id
+  const destinationAccountLedger = args.destinationAccount.asset.ledger
+
+  const { sourceAmount, destinationAmount, timeout } = args
+
+  if (sourceAccountRef === destinationAccountRef) {
+    return TransferError.SameAccounts
+  }
+
+  if (sourceAmount <= 0n) {
+    return TransferError.InvalidSourceAmount
+  }
+
+  if (destinationAmount !== undefined && destinationAmount <= 0n) {
+    return TransferError.InvalidDestinationAmount
+  }
```

Comment:
What if this and tigerbeetle used a shared validation function?

Reply:
Similarly, how much of the logic below could exist in `accounting/service.ts`?
Could there be a shared function that takes tigerbeelte/psql specific `addTransfer`, etc?

Reply:
> Could there be a shared function that takes tigerbeelte/psql specific addTransfer, etc?
Went down a fun path with this one, I'm giving it a go